src/Mapping/Event/Adapter/ORM.php

```diff
@@ -95,6 +95,18 @@ public function getObjectManager()
             throw new \LogicException(sprintf('Event args must be set before calling "%s()".', __METHOD__));
         }
 
+        if (\method_exists($this->args, 'getObjectManager')) {
+            return $this->args->getObjectManager();
+        }
+
+        @trigger_error(sprintf(
+            'Calling "%s()" on an event of type "%s" is deprecated since gedmo/doctrine-extensions 3.x'
+            .' and will throw a "%s" error in version 4.0.',
+            __METHOD__,
+            get_class($this->args),
+            \TypeError::class
```

Comment:
Should this be `\Error::class`?

Reply:
I think a TypeError is fine if we look at it as the $args class (type) not implementing the getObjectManager method. If we look at it like "the type $args::class must implement getObjectManager for version 4.0" then it should be \Error.

Reply:
If there won't be an `instanceof` check here, I guess we can't even be sure if`\Error` will be thrown. If that is the case, I guess we should say something like "it will be not supported" instead.

Reply:
No instanceof check is required, calling a method that doesn't exist throws an \Error.

Fatal error: Uncaught Error: Call to undefined method OldEventArgs::getObjectManager()

Reply:
Can we maybe add instanceof checks with `LifecycleEventArgs`, `LoadClassMetadataEventArgs`, `ManagerEventArgs` and `OnClearEventArgs`? Maybe they should all extend from `ManagerEventArgs`  in `doctrine/persistence` 🤔 ? For the `getObjectManager()` call I mean, or any other solution?

It would be nice to finally move this forward.

Reply:
I don't see any need to do that. The \method_exists call will make sure that it's fully backwards compatible for now.

If anything, that would be a discussion for later when we remove the  \method_exists call and directly call `getObjectManager()`. But even then, I don't see a point in checking the type of the args - all we care about is that they abide by Doctrines interfaces and implement `getObjectManager()` instead of `getEntityManager()` (that we currently also call without checking).